cli/command/registry/login.go

```diff
@@ -108,7 +108,7 @@ func runLogin(ctx context.Context, dockerCli command.Cli, opts loginOptions) err
 	// if we failed to authenticate with stored credentials (or didn't have stored credentials),
 	// prompt the user for new credentials
 	if err != nil || authConfig.Username == "" || authConfig.Password == "" {
-		response, err = loginUser(ctx, dockerCli, opts, authConfig.Username, serverAddress)
+		response, err = loginUser(ctx, dockerCli, opts, authConfig.Username, authConfig.ServerAddress)
```

Comment:
Do we have some test that we can add these cases on and/or mock something to have a regression test?

Reply:
I'll add them, I just wanted to do a PR quickly in case we wanted to release. I'll do it separately in a bit.

Reply:
Added some/refactored the tests there, PTAL if you get a sec